controllers/cluster_controller.go

```diff
@@ -219,6 +219,9 @@ func (r *ClusterReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 					"failed to delete %v %q for Cluster %q in namespace %q",
 					obj.GroupVersionKind(), obj.GetName(), cluster.Name, cluster.Namespace)
 			}
+		default:
+			logger.Info("Cluster still has descendants - need to requeue", "controlPlaneRef", cluster.Spec.ControlPlaneRef.Name)
+			return ctrl.Result{RequeueAfter: deleteRequeueAfter}, nil
```

Comment:
I don't think we can do this here - we have code in the KCP controller that defers deletion of the control plane machines if there are any non-control plane machines still around. Making this change here would cause a deadlock, I think.

Reply:
I forget why the ControlPlaneRef is handled first, shouldn't that be the last object to be deleted?

Reply:
I'm sure we must have had a reason for doing it first...

Reply:
Original PR https://github.com/kubernetes-sigs/cluster-api/pull/2094, trying to dig through the comments, I might just test it moving it down though